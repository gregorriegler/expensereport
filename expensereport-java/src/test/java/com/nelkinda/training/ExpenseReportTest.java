package com.nelkinda.training;

import org.approvaltests.ApprovalUtilities;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.util.Collections;
import java.util.Date;

public class ExpenseReportTest {

    private final ByteArrayOutputStream outputStream = new ApprovalUtilities().writeSystemOutToStringBuffer();
    private final ExpenseReport expenseReport = new ExpenseReport();

    @Test
    public void printEmptyReport() {
        expenseReport.printReport(Collections.emptyList(), new Date(0));

        Approvals.verify(outputStream.toString());
    }

    @Test
    public void printDinner() {
        expenseReport.printReport(Collections.singletonList(createExpense(ExpenseType.DINNER, 10)), new Date(0));

        Approvals.verify(outputStream.toString());
    }

    @Test
    public void printBreakfast() {
        expenseReport.printReport(Collections.singletonList(createExpense(ExpenseType.BREAKFAST, 10)), new Date(0));

        Approvals.verify(outputStream.toString());
    }

    @Test
    public void printCarRental() {
        expenseReport.printReport(Collections.singletonList(createExpense(ExpenseType.CAR_RENTAL, 10)), new Date(0));

        Approvals.verify(outputStream.toString());
    }

    private Expense createExpense(ExpenseType expenseType, int amount) {
        Expense expense = new Expense();
        expense.type = expenseType;
        expense.amount = amount;
        return expense;
    }

}
