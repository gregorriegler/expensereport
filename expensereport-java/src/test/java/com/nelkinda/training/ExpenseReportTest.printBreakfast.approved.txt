Expenses Thu Jan 01 01:00:00 CET 1970
Breakfast	10	 
Meal expenses: 10
Total expenses: 10
